persistence/test/account_test.go

```diff
@@ -355,15 +369,19 @@ func TestGetAllPools(t *testing.T) {
 	db := NewTestPostgresContext(t, 0)
 
 	updatePool := func(db *persistence.PostgresContext, pool *coreTypes.Account) error {
-		return db.AddPoolAmount(pool.GetAddress(), "10")
+		addrBz, err := hex.DecodeString(pool.Address)
+		require.NoError(t, err)
+		return db.AddPoolAmount(addrBz, "10")
 	}
 
-	getAllActorsTest(t, db, db.GetAllPools, createAndInsertNewPool, updatePool, 7)
+	// -1 because we don't count the "unspecified" pool (Pools_POOLS_UNSPECIFIED)
+	initialCount := len(coreTypes.Pools_value) - 1
+	getAllActorsTest(t, db, db.GetAllPools, createAndInsertNewPool, updatePool, initialCount)
 }
 
 func TestPoolsUpdatedAtHeight(t *testing.T) {
 	db := NewTestPostgresContext(t, 0)
-	numPoolsInTestGenesis := 7
+	numPoolsInTestGenesis := len(coreTypes.Pools_value) - 1 // -1 because we don't count the "unspecified" pool (Pools_POOLS_UNSPECIFIED
```

Comment:
```suggestion
	numPoolsInTestGenesis := len(coreTypes.Pools_value) - 1 // -1 because we don't count the "unspecified" pool (Pools_POOLS_UNSPECIFIED)
```